# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannelParent.cpp
# Commit: 8575cd92e1d8
# Full Hash: 8575cd92e1d8cd783e6cca3f6feadef11f0c4cc8
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-03 09:24:59
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 8: Add list of open channels from OngoingEarlyHints to RedirectToRealChannelArgs r=necko-reviewers,valentin,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161179
# ==============================================================================

diff -r ce8f0a8fa688 -r 8575cd92e1d8 netwerk/ipc/DocumentChannelParent.cpp
--- a/netwerk/ipc/DocumentChannelParent.cpp	Fri Dec 02 16:15:55 2022 +0000
+++ b/netwerk/ipc/DocumentChannelParent.cpp	Fri Dec 02 16:15:56 2022 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/dom/CanonicalBrowsingContext.h"
 #include "mozilla/dom/ClientInfo.h"
 #include "mozilla/dom/ContentParent.h"
+#include "mozilla/net/NeckoChannelParams.h"
 #include "nsDocShellLoadState.h"
 
 extern mozilla::LazyLogModule gDocumentChannelLog;
@@ -96,7 +97,8 @@
         // PDocumentChannel::RedirectToRealChannelPromise given as parameter.
         auto promise = self->RedirectToRealChannel(
             std::move(aResolveValue.mStreamFilterEndpoints),
-            aResolveValue.mRedirectFlags, aResolveValue.mLoadFlags);
+            aResolveValue.mRedirectFlags, aResolveValue.mLoadFlags,
+            std::move(aResolveValue.mEarlyHints));
         // We chain the promise the DLL is waiting on to the one returned by
         // RedirectToRealChannel. As soon as the promise returned is resolved
         // or rejected, so will the DLL's promise.
@@ -135,7 +137,8 @@
 DocumentChannelParent::RedirectToRealChannel(
     nsTArray<ipc::Endpoint<extensions::PStreamFilterParent>>&&
         aStreamFilterEndpoints,
-    uint32_t aRedirectFlags, uint32_t aLoadFlags) {
+    uint32_t aRedirectFlags, uint32_t aLoadFlags,
+    nsTArray<EarlyHintConnectArgs>&& aEarlyHints) {
   if (!CanSend()) {
     return PDocumentChannelParent::RedirectToRealChannelPromise::
         CreateAndReject(ResponseRejectReason::ChannelClosed, __func__);
@@ -143,7 +146,8 @@
   RedirectToRealChannelArgs args;
   mDocumentLoadListener->SerializeRedirectData(
       args, false, aRedirectFlags, aLoadFlags,
-      static_cast<ContentParent*>(Manager()->Manager()));
+      static_cast<ContentParent*>(Manager()->Manager()),
+      std::move(aEarlyHints));
   return SendRedirectToRealChannel(args, std::move(aStreamFilterEndpoints));
 }
 